wpengine/faustjs
# App Router: `getAuthClient` and `faustRouteHandler`

## Tasks


## Description

This PR sets up the new API Router but for App Router, called `faustRouteHandler`, and creates the `getAuthClient` function. This is a function, when authenticated, will return an Apollo client with the access token configured as the `Authorization` bearer token. When unauthenticated, the client returns `null`. 



## Related Issue(s):



## Testing

**Additional tickets will be created for proper auth methods like login/logout. But for the scope of this ticket, we will set the tokens manually.**

1. `npm install` from the monorepo root
2. `npm run build` from the monorepo root to build the packages
3. `npm run dev -w @faustwp/app-router-example` to run the example project
4. Navigate to `/gated-content` and notice the "You must be authenticated" text.
5. Go to your WordPress instance and open the GraphiQL IDE
6. Use the following mutation to generate an authorization code (this code is valid for 60 seconds):
    ```graphql
    mutation MyMutation {
      generateAuthorizationCode(input: {username: "admin", password: "admin"}) {
        clientMutationId
        code
        error
      }
    }
    ```
7. Encode the `code` for URLs. You can do that here: https://www.urlencoder.org
8.  With the encoded authorization code, visit `localhost:3000/api/faust/token?code=xxxx`, you should see a response with access tokens and refresh tokens.
9. Visiting the above endpoint will set the valid refresh token in the cookie. You can double check it was set by opening your developer tools > storage > cookies.
10. Navigation back to http://localhost:3000/gated-content and notice your authenticated viewer name and your posts 

Video demonstration of testing instructions:

https://github.com/wpengine/faustjs/assets/5946219/b4f64d57-a5d5-4cab-aa5a-af3fa940d3f0



## Screenshots



## Documentation Changes



## Dependant PRs
